### PR TITLE
Support Python3

### PR DESCRIPTION
**Pull request by jslee02**
This PR:
* merges the upstream master
* adds CMake variable `BOOST_NUMPY_PYTHON_VERSION` to correctly support Python2 and Python3 like https://github.com/personalrobotics/dartpy/pull/36

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -45,7 +49,10 @@ set(CMAKE_MACOSX_RPATH 1)
 
 if(${PYTHON_VERSION_STRING} GREATER 3.0)
   message(STATUS "Using Python3")
-  find_package(Boost COMPONENTS python3 REQUIRED)
+  find_package(Boost COMPONENTS python-py${PYTHON_VERSION_MAJOR}${PYTHON_VERSION_MINOR})
+  if (NOT Boost_FOUND)
+    find_package(Boost COMPONENTS python3 REQUIRED)
+  endif()
```
Comment (psigen) [inline CMakeLists.txt]: @jslee02 did you just come up with this snippet to find the Boost Python component, or is it from somewhere else?  It seems pretty arbitrary...

File: .travis.yml [hunk 2]
```diff
@@ -18,5 +18,4 @@ script:
   - cd ../libs/numpy/doc
   - make html
   - cd ../../../
-  - if [ "$PYTHON" == "/usr/bin/python" ]; then scons; fi
-
+  - if [ "$TRAVIS_PYTHON_VERSION" == "2.7" ]; then scons; fi
```
Comment (psigen) [inline .travis.yml]: Does this mean we don't run `scons` at all if python 3.5 is selected?

**Review by psigen — COMMENTED** [review]
I'm not sure I understand the change to the SCONS command here, the rest looks ok.

File: .travis.yml (hunk 2 above)
Comment (jslee02) [reply .travis.yml]: I don't know why, but it seems so. :thinking:  [It's just from the upstream](https://github.com/ndarray/Boost.NumPy/commit/4b14f7731c8cccd63a41adecd95642a302ccca3b).

File: CMakeLists.txt (hunk 1 above)
Comment (jslee02) [reply CMakeLists.txt]: I can't find the original snippet now, but it's from several repositories.
 Here are some of them:
* https://github.com/halide/Halide/blob/master/python_bindings/CMakeLists.txt#L7-L27
* https://pypkg.com/pypi/doom-py/f/doom_py/CMakeLists.txt
* https://sourceforge.net/p/netgen-mesher/git/ci/b3451ed2e473f7308f6cec6b0f9f058183e4aaba/

Yeah, I was surprised at this pattern too. :disappointed:

**Comment by psigen** [comment]
Did that change get reverted? I don’t see it in their master. It looks like
it might have been a temporary fix for some build issue.

https://github.com/ndarray/Boost.NumPy/blob/master/.travis.yml

**Comment by jslee02** [comment]
I believe [these lines](https://github.com/ndarray/Boost.NumPy/blob/master/.travis.yml#L18-L22) are equivalent to `if [ "$TRAVIS_PYTHON_VERSION" == "2.7" ]; then scons; fi` because the python and python3 versions on Trusty are 2.7 and 3.4, respectively.

**Comment by psigen** [comment]
Ah, ok that makes sense, thanks.

>

**Pull request merged**